backend/src/main/java/com/capstone/moneytree/facade/MarketInteractionsFacade.java

```diff
@@ -211,22 +209,33 @@ public void disconnectFromStream(String userId) {
     */
    private AlpacaStreamListener createStreamListener(String userId, SimpMessagingTemplate messageSender,
                                                      AlpacaStreamMessageType... messageType) {
+
+
       return new AlpacaStreamListenerAdapter(messageType) {
          @Override
          public void onStreamUpdate(AlpacaStreamMessageType streamMessageType, AlpacaStreamMessage streamMessage) {
+
             TradeUpdateMessage tradeMessage = (TradeUpdateMessage) streamMessage;
             TradeUpdate tradeUpdate = tradeMessage.getData();
-
-            if (streamMessageType == AlpacaStreamMessageType.TRADE_UPDATES &&
-                    !tradeUpdate.getOrder().getStatus().equals("filled") &&
-                    madeDao.findByTransactionId(transactionDao.findByClientOrderId(tradeUpdate.getOrder().getClientOrderId()).getId())
-                            .getUser().getId().toString().equals(userId) && tradeUpdate.getEvent().equals("fill")) {
-
-               messageSender.convertAndSend("/queue/user-" + userId,
-                       tradeUpdate.getOrder().getClientOrderId());
-               sendOrderCompletedEmail(userId, tradeUpdate);
-               updateTransactionStatus(tradeUpdate.getOrder().getClientOrderId(), tradeUpdate.getOrder().getFilledAvgPrice(), tradeUpdate.getPrice());
-               LOGGER.info("Order filled by user id {}", userId);
+            String clientOrderId = tradeUpdate.getOrder().getClientOrderId();
+
+            updateTracker.putIfAbsent(clientOrderId, false);
+
+            if(tradeUpdate.getEvent().equals("fill")) {
+               Transaction transaction = transactionDao.findByClientOrderId(clientOrderId);
+               while (transaction == null ) {
+                  transaction = transactionDao.findByClientOrderId(clientOrderId);
+               }
+               String userIdForOrder = madeDao.findByTransactionId(transaction.getId()).getUser().getId().toString();
+               if (streamMessageType == AlpacaStreamMessageType.TRADE_UPDATES && !updateTracker.get(clientOrderId) && userIdForOrder.equals(userId)
```

Comment:
Just did some research on why the streams are not being removed when disconnecting and it I haven't found anything relavant. I'll try to search more. I also just realized that this if statement isn't tested at all. I think this will require another story to create an integration test to properly test it. It isn`t a top priority right now.